src/login/logind-core.c

```diff
@@ -393,6 +393,28 @@ int manager_get_session_by_pidref(Manager *m, const PidRef *pid, Session **ret)
         return !!s;
 }
 
+int manager_get_session_by_leader(Manager *m, const PidRef *pid, Session **ret) {
```

Comment:
can you update _by_pidref variant to call into this first and check for != 0 for early return?

Reply:
Done.